### ISSUE TITLE
Error when plotting features after data integration

### ISSUE THREAD

**Issue by ccruizm** [satijalab/seurat#3786]
Good day,

I am having problems plotting some genes after data integration. I do not understand what the problem can be since many other genes are plotted nicely but for some specific ones, I get the next error message (using `Seurat v3.2.2 ` or `v4` in `R v4`):
```
ERROR while rich displaying an object: Error: Discrete value supplied to continuous scale

Traceback:
1. FUN(X[[i]], ...)
2. tryCatch(withCallingHandlers({
 .     if (!mime %in% names(repr::mime2repr)) 
 .         stop("No repr_* for mimetype ", mime, " in repr::mime2repr")
 .     rpr <- repr::mime2repr[[mime]](obj)
 .     if (is.null(rpr)) 
 .         return(NULL)
 .     prepare_content(is.raw(rpr), rpr)
 . }, error = error_handler), error = outer_handler)
3. tryCatchList(expr, classes, parentenv, handlers)
4. tryCatchOne(expr, names, parentenv, handlers[[1L]])
5. doTryCatch(return(expr), name, parentenv, handler)
6. withCallingHandlers({
 .     if (!mime %in% names(repr::mime2repr)) 
 .         stop("No repr_* for mimetype ", mime, " in repr::mime2repr")
 .     rpr <- repr::mime2repr[[mime]](obj)
 .     if (is.null(rpr)) 
 .         return(NULL)
 .     prepare_content(is.raw(rpr), rpr)
 . }, error = error_handler)
7. repr::mime2repr[[mime]](obj)
8. repr_text.default(obj)
9. paste(capture.output(print(obj)), collapse = "\n")
10. capture.output(print(obj))
11. evalVis(expr)
12. withVisible(eval(expr, pf))
13. eval(expr, pf)
14. eval(expr, pf)
15. print(obj)
16. print.patchwork(obj)
17. build_patchwork(plot, plot$layout$guides %||% "auto")
18. lapply(x$plots, plot_table, guides = guides)
19. FUN(X[[i]], ...)
20. plot_table.ggplot(X[[i]], ...)
21. ggplotGrob(x)
22. ggplot_gtable(ggplot_build(x))
23. ggplot_build(x)
24. ggplot_build.ggplot(x)
25. lapply(data, scales_train_df, scales = npscales)
26. FUN(X[[i]], ...)
27. lapply(scales$scales, function(scale) scale$train_df(df = df))
28. FUN(X[[i]], ...)
29. scale$train_df(df = df)
30. f(..., self = self)
31. self$train(df[[aesthetic]])
32. f(..., self = self)
33. self$range$train(x)
34. f(..., self = self)
35. scales::train_continuous(x, self$range)
36. stop("Discrete value supplied to continuous scale", call. = FALSE)
```
What do you think the issue might be? Thanks in advance for your help!

**Comment by andrewwbutler** [comment]
Hi,

What kind of plot are you trying to make here? Can you provide the code you ran, ideally as a reproducible example?

**Comment by ccruizm** [comment]
Solved in #3768
